File: docs/versioned_docs/version-1.11.0/API-Reference/python-examples/api-openai-responses/continue-conversations-with-response-and-session-ids-2.py
# Continuation requests use the same endpoint; add `previous_response_id` from a prior
# response's `id` field. The bootstrap flow is empty; use a Playground flow with ChatInput +
# ChatOutput for a full run. Same first message as continue-conversations-with-response-and-session-ids.py.

import os

import requests

base = os.environ.get("LANGFLOW_URL") or os.environ.get("LANGFLOW_SERVER_URL", "")
flow_id = os.environ.get("FLOW_ID", "")
api_key = os.environ.get("LANGFLOW_API_KEY", "")

url = f"{base}/api/v1/responses"

headers = {
    "x-api-key": api_key,
    "Content-Type": "application/json",
}

payload = {
    "model": flow_id,
    "input": "Hello, my name is Alice",
    "stream": False,
}

response = requests.post(url, headers=headers, json=payload, timeout=120)
response.raise_for_status()

print(response.text)

File: docs/versioned_docs/version-1.12.0/API-Reference/javascript-examples/api-openai-responses/continue-conversations-with-response-and-session-ids-2.js
const url = `http://${process.env.LANGFLOW_SERVER_URL ?? ""}/api/v1/responses`;

const options = {
  method: 'POST',
  headers: {
    "x-api-key": `${process.env.LANGFLOW_API_KEY ?? ""}`,
    "Content-Type": `application/json`,
  },
  body: JSON.stringify({
  "model": "ced2ec91-f325-4bf0-8754-f3198c2b1563",
  "input": "What's my name?",
  "previous_response_id": "c45f4ac8-772b-4675-8551-c560b1afd590"
}),
};

fetch(url, options)
  .then(async (response) => {
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}`);
    }
    const text = await response.text();
    console.log(text);
  })
  .catch((error) => console.error(error));
